rust/hyperlane-core/src/traits/signing.rs

```diff
@@ -80,6 +84,20 @@ pub struct SignedType<T: Signable> {
     pub signature: Signature,
 }
 
+impl<T: Signable + Serialize> Serialize for SignedType<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("SignedType", 3)?;
+        state.serialize_field("value", &self.value)?;
+        state.serialize_field("signature", &self.signature)?;
+        let sig = <[u8; 65]>::from(self.signature);
```

Comment:
Minor nit to be more readable
```suggestion
        let sig: [u8; 65] = self.signature.into();
```